src/coreclr/nativeaot/Runtime/thread.inl

```diff
@@ -56,3 +56,18 @@ inline bool Thread::IsInCantAllocStressLogRegion()
 {
     return m_cantAlloc != 0;
 }
+
+struct CantAllocHolder
```

Comment:
I think, maybe this should be debug-only code. This has now been reduced to ensuring that we do not allocate stresslog buffers in 3 or 4 fairly trivial methods.

We just need to be sure that those methods do not use stress log. I think we could rely on asserts.